pkg/webhook/cluster/nutanix_validator.go

```diff
@@ -132,31 +132,55 @@ func validateControlPlaneFailureDomainConfig(clusterConfig *variables.ClusterCon
 		"machineDetails",
 	)
 
-	if clusterConfig.ControlPlane != nil &&
-		clusterConfig.ControlPlane.Nutanix != nil &&
-		len(clusterConfig.ControlPlane.Nutanix.FailureDomains) == 0 {
+	if clusterConfig.ControlPlane != nil && clusterConfig.ControlPlane.Nutanix != nil {
 		machineDetails := clusterConfig.ControlPlane.Nutanix.MachineDetails
-		if machineDetails.Cluster == nil || (!machineDetails.Cluster.IsName() && !machineDetails.Cluster.IsUUID()) {
-			fldErrs = append(fldErrs, field.Required(
-				fldPath.Child("cluster"),
-				"\"cluster\" must be set when failureDomains are not configured.",
-			))
-		}
+		hasFailureDomains := len(clusterConfig.ControlPlane.Nutanix.FailureDomains) > 0
+		hasCluster := machineDetails.Cluster != nil &&
+			(machineDetails.Cluster.IsName() || machineDetails.Cluster.IsUUID())
+		hasSubnets := len(machineDetails.Subnets) > 0
```

Comment:
nit: there are 2 places that we have the same checks, we could add a Method on `NutanixMachineDetails` to do this